### DISCUSSION TITLE
Set text size

### DISCUSSION THREAD

**Discussion by memeplex** [cetz-package/cetz#669] [Q&A] [answered]
Is there any way to set text size inside a canvas, other than calling `text()` explicitly?

There is no relevant style and `set text(size: ...)` inside the canvas will fail because its result is of incorrect type.

Thanks!

**Comment by fenjalien** [comment, marked as answer]
You can `set text` outside of the canvas but if you want to change it within the canvas you would have to set it for each `content` function due to how scopes work in Typst.

To save a bit of time you could write a wrapper function that applies the styling you want:
```
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *

  let my-content(a, b, ..rest) = {
    content(
      a,
      {
        set text(size: 20pt)
        b
      },
      ..rest
    )
  }

  content((), [hello])
  my-content((rel: (2,0)), [hello])
})
```

**Reply by gitAlwin** [reply]
Hi, the following is perhaps a noob question but I can't get it work just for a single use of the `content` function. I.e. how do I change the font size of a single `content` function without creating a helper function for it. I tried 
```
  content((0, 0), {set text(size: 50pt)}, [b])
```
```
  content((0, 0), {set text(size: 50pt)[b]})
```
and
```
  content((0, 0), {set text(size: 50pt)b})
```
and variations on these but nothing was successful. What would be the right syntax for this? I did try to search the manual but couldn't find anything that describes such a situation.

Thank you for any help

**Reply by johannes-wolf** [reply]
Normal Typst syntax: `content((0,0), text(size: 50pt)[b])`
- or -
`content((0,0), { set text(size: 50pt); [b] })` (you need a space after the closing paren).

**Reply by gitAlwin** [reply]
Hello Johannes, 

Thank you very much for your answer, it resolved my issue. But including the space after the closing parenthesis as in your second example actually yields an error saying `Expected semicolon or line break`. Not including that space works though: `content((2,0), {text(size: 50pt)[b]})` works. Just a detail. Anyway, thank you very much for answering my question and for creating this package together with [fenjalien](https://github.com/fenjalien). I'm coming from TikZ and am learning CeTZ now, much respect for you and [fenjalien](https://github.com/fenjalien) creating it!

edit: also love CeTZ-plot ❤️

**Reply by johannes-wolf** [reply]
Thank you! :)
(Edited my answer, I wanted to write `{set text(size: 50pt); [ ... ]}`.)
